jgrapht-core/src/main/java/org/jgrapht/alg/cycle/ChordalityInspector.java

```diff
@@ -0,0 +1,211 @@
+ * order isn't unique. Both maximum cardinality search and lexicographical breadth-first search produces this order.
+ * As a result, chordality inspection can be performed in two way. Here it is done by default via MCS,
+ * because it outperforms LexBFS by a constant factor in practice. Nevertheless, lexicographical breadth-first
+ * order can be obtained via {@link ChordalityInspector#getLexicographicalBfsOrder()}.
+ * <p>
+ * Both lexicographical BFS and maximum cardinality search run in O(|V| + |E|). Checking whether given order is the perfect elimination order
+ * via {@link ChordalityInspector#isPerfectEliminationOrder(List)} takes O(|V| + |E|) as well. So,
+ * overall time complexity of the method {@link ChordalityInspector#isChordal()} is O(|V| + |E|).
+ *
+ * @param <V> the graph vertex type.
+ * @param <E> the graph edge type.
+ * @author Timofey Chudakov
+ * @see LexicographicalBfsIterator
+ * @see MaximumCardinalityIterator
+ */
+public class ChordalityInspector<V, E> {
+    /**
+     * The inspected graph.
+     */
+    private Graph<V, E> graph;
+    /**
+     * Contains true if the graph is chordal, otherwise false. Is null before the first call to the
+     * {@link ChordalityInspector#isChordal()}.
+     */
+    private Boolean chordal = null;
+
+    /**
+     * Vertices of the {@code graph} in a maximum cardinality order.
+     */
+    private List<V> mcsOrder;
+    /**
+     * Vertices of the {@code graph} in a lexicographical breadth-first order.
+     */
+    private List<V> lexBfsOrder;
+
+
+    /**
+     * Creates a chordality inspector for {@code graph}.
+     *
+     * @param graph the graph for which a chordality inspector to be created.
+     */
+    public ChordalityInspector(Graph<V, E> graph) {
+        this.graph = Objects.requireNonNull(graph);
+        if (graph.getType().isDirected()) {
+            this.graph = new AsUndirectedGraph<>(graph);
+        }
+
+    }
+
+    /**
+     * Checks whether the inspected graph is chordal. Note: Every time this method is invoked, LexBFS order is recomputed.
+     *
+     * @return true if this graph is chordal, otherwise false.
+     */
+    public boolean isChordal() {
+        if (chordal == null) {
+            mcsOrder = mcs();
+            chordal = isPerfectEliminationOrder(mcsOrder);
+        }
+        return chordal;
+    }
+
+
+    /**
+     * Lazily computes lexicographical breadth-first order of the inspected graph.
+     * Note: chordality of the {@code graph} is tested with the {@link MaximumCardinalityIterator} by default.
+     * As the result, LexBFS order is computed only after the first invocation of this method.
+     * If the case of modifications, this method returns order for the most recent version of the {@code graph}.
+     *
+     * @return the vertices of the {@code graph} in the lexicographical breadth-first order.
+     */
+    public List<V> getLexicographicalBfsOrder() {
```

Comment:
This method is not needed anymore, the user can use directly the Iterator.

Reply:
I have a question. There is no need to save computed order at all or ChordalityInspector can save computed ordering in order to, for example, return it as a perfect elimination order after the graph turned out to be chordal?

Reply:
Better to cache the computed order and return it as a perfect elimination order.

Reply:
Thanks